src/odemis/gui/util/img.py

```diff
@@ -1197,30 +1197,29 @@ def clip_data_window(hrange, vrange, xd, yd):
 
 def spectrum_to_export_data(proj, raw, vp=None):
     """
-    Creates either raw or WYSIWYG representation for the spectrum data plot
-
-    proj (DataProjection): DataProjection of the spectrum to export
-    raw (boolean): if True returns raw representation
-    vp (Viewport): Viewport object
-
+    Creates either the raw or the representation as shown in the viewport in the GUI
+    (WYSIWYG) of the spectrum data plot for export.
+    :param proj: (SinglePointSpectrumProjection) A spectrum projection.
+    :param raw: (boolean) If True, returns raw representation.
+    :param vp: (Viewport or None) The viewport selected for export to get the data displayed
+               in the viewport.
+    :returns: (model.DataArray) The data array to export.
     TODO: Instead of a viewport, pass a gui.model.StreamView. That (special) view
     would have the corresponding VAs hrange and vrange.
     The NavigablePlotViewport would just read/write them,
     whenever the user moves or zooms around.
-
-    returns (model.DataArray)
     """
-    spec = proj.projectAsRaw()
-    if spec is None:
-        raise LookupError("No pixel selected to pick a spectrum")
-    spectrum_range, unit = spectrum.get_spectrum_range(spec)
-
-    if raw:
-        if unit == "m":
-            spec.metadata[model.MD_WL_LIST] = spectrum_range
-        spec.metadata[model.MD_ACQ_TYPE] = model.MD_AT_SPECTRUM
-        return spec
-    else:
+    if raw:  # csv
```

Comment:
In the case of the point spectrum/chronogram, the `.image` is the same as `.projectAsRaw()`. So that previous code wasn't incorrect. However, I think it's a little better now, as it makes all the 4 functions follow the same pattern.

Reply:
ok good :)